src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentMarkupEncodingPass.cs

```diff
@@ -64,7 +68,68 @@ public override void VisitHtml(HtmlContentIntermediateNode node)
                             return;
                         }
                     }
+
+                    token.Content = DecodeHtmlEntities(token.Content);
+                }
+            }
+
+            private string DecodeHtmlEntities(string content)
```

Comment:
I know this and the below method can be combined and simplified a little but I decided to go with a little more readability.